src/test/ui/closures/2229_closure_analysis/diagnostics/liveness.rs

```diff
@@ -0,0 +1,62 @@
+// check-pass
+#![feature(capture_disjoint_fields)]
+//~^ WARNING: the feature `capture_disjoint_fields` is incomplete
+#![allow(unreachable_code)]
+
+#[derive(Debug)]
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+pub fn f() {
+    let mut c = Point{ x:1, y:0 };
+
+    // Captured by value, but variable is dead on entry.
+    (move || {
+        c.x = 1; //~ WARN value captured by `c` is never read
+        println!("{}", c.x);
+    })();
+
+    // Read and written to, but never actually used.
+    (move || {
+        c.x += 1; //~ WARN unused variable: `c`
```

Comment:
```suggestion
        // This will not trigger a warning for unused variable as c.x will be treated as a Non-tracked place
        c.x += 1;
        a += 1; //~ WARN unused variable: `a`
```

`c.x` will not trigger a warning because `c.x` is treated differently than a root variable. As you can [see here](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L1148-L1162) places are treated differently based on whether it is a root variable  or a non-root variable.

The typical flow is:

1. `ACC_READ` & `ACC_WRITE` required for the warning to be triggered will be assigned [here](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L957)
2. However, for a non-root variable the expr kind looks like a `Field` that contains an expression kind that is a `Path` instead of just `Path`, so we will never actually assign the above accesses. [see here](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L1178-L1190)
3. We will however [continue propagating](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L959) through the left hand side of this expression, which is just continuing to [propagate through the expr](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L1172). As a result, we actually assign the access `ACC_READ` & `ACC_USES`. [See here](https://github.com/sexxi-goose/rust/blob/master/compiler/rustc_passes/src/liveness.rs#L819-L821)

Conclusion, since the accesses are not what we initially expected, the warning will never be triggered. This "issue" shouldn't be addressed here, so just modify the test case so it uses the root variable instead, this way we know this change didn't break the current expectation.